pkg/route/hostassignment/assignment.go

```diff
@@ -137,7 +137,7 @@ func certificateChangeRequiresAuth(route, older *routev1.Route, opts route.Route
 			a.Key != b.Key
 
 		if opts.AllowExternalCertificates {
-			if route.Spec.TLS.ExternalCertificate != nil || older.Spec.TLS.ExternalCertificate != nil {
+			if route.Spec.TLS.ExternalCertificate != nil {
```

Comment:
I suppose this change is responsible for the note from the PR description:

```
Note: since the state of the external secret cannot be verified, even keeping the secret name unchanged, requires permission check.
```

I think this code deserves a comment with similar description.

Reply:
The function godoc has a similar description. Let me know if we need to include anything else. 

```go
// Note: If (newer/updated) route uses externalCertificate, this function always returns true, as we cannot definitively verify if
// the content of the referenced secret has been modified. Even if the secret name remains the same,
// we must assume that the secret content is changed, necessitating authorization.
```